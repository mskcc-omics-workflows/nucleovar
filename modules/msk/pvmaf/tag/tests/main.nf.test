nextflow_process {

    name "Test Process PVMAF_TAG"
    script "../main.nf"
    process "PVMAF_TAG"
    config "./nextflow.config"

    tag "modules"
    tag "modules_nfcore"
    tag "modules_msk"
    tag "pvmaf"
    tag "pvmaf/tag"

    test("chr22mafstandard - msk - traceback") {


        when {
            process {
                """input[0] = [
                            [ id:'chr22', patient:'test'], // meta map
                            file(params.test_data_mskcc['chr22']['chr22mafstandard'], checkIfExists: true),
                        ]
                input[1] = 'traceback'
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("chr22mafstandard - msk - traceback - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                            [ id:'chr22', patient:'test'], file(params.test_data_mskcc['chr22']['chr22mafstandard'], checkIfExists: true),
                        ]
                input[1] = 'traceback'
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
