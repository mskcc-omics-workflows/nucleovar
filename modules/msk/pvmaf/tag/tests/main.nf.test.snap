{
    "chr22mafstandard - msk - traceback": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "chr22",
                            "patient": "test"
                        },
                        "test_traceback.maf:md5,23ca5bc6918acfd84b98fdce78bc008b"
                    ]
                ],
                "1": [
                    "versions.yml:md5,1f95a3b0ed06d92f31b49e3b61331d23"
                ],
                "maf": [
                    [
                        {
                            "id": "chr22",
                            "patient": "test"
                        },
                        "test_traceback.maf:md5,23ca5bc6918acfd84b98fdce78bc008b"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,1f95a3b0ed06d92f31b49e3b61331d23"
                ]
            }
        ],
        "timestamp": "2024-02-29T12:21:12.874605"
    },
    "chr22mafstandard - msk - traceback - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "chr22",
                            "patient": "test"
                        },
                        "test_traceback.maf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,1f95a3b0ed06d92f31b49e3b61331d23"
                ],
                "maf": [
                    [
                        {
                            "id": "chr22",
                            "patient": "test"
                        },
                        "test_traceback.maf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,1f95a3b0ed06d92f31b49e3b61331d23"
                ]
            }
        ],
        "timestamp": "2024-02-29T12:21:24.705074"
    }
}